mesonbuild/compilers/swift.py

```diff
@@ -49,6 +50,16 @@ def get_werror_args(self) -> T.List[str]:
     def get_dependency_gen_args(self, outtarget: str, outfile: str) -> T.List[str]:
         return ['-emit-dependencies']
 
+    def get_dependency_link_args(self, dep: 'Dependency') -> T.List[str]:
+        result = []
+        for arg in dep.get_link_args():
+            if arg.startswith("-Wl,"):
+                for flag in arg[4:].split(","):
+                    result += ["-Xlinker", flag]
```

Comment:
This isn't right. But calling `.split()` here flag becomes a `List[str]`, that means that result ends up with `['-Xlinker', ['-linkarg']]`, which is not a string.